### ISSUE TITLE
swissknife inject command crashes my wear app, also injected views are null

### ISSUE THREAD

**Issue by rvanderwerf** [Arasthel/SwissKnife#38]
I have a fully functioning wear app trying to add SwissKnife too, but it barfs as soon as the inject method is called on SwissKnife. Also my injected views are null.

code that crashes:
https://gist.github.com/rvanderwerf/1dab5d2fbf4a469d700c
@InjectView(value=R.id.pager)
    GridViewPager pager

    @InjectView(value=R.id.page_indicator)
    DotsPageIndicator dotsPageIndicator

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        SwissKnife.inject(this)

        pager = (GridViewPager) findViewById(R.id.pager)
        dotsPageIndicator = (DotsPageIndicator) findViewById(R.id.page_indicator)
        dotsPageIndicator.setPager(pager)

I get this exception: 

Caused by: groovy.lang.MissingMethodException: No signature of method: com.tutosandroidfrance.wearsample.MainActivity.injectViews() is applicable for argument types: (com.tutosandroidfrance.wearsample.MainActivity) values: [com.tutosandroidfrance.wearsample.MainActivity@133ab8db]
at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.unwrap(ScriptBytecodeAdapter.java:58)
at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:54)
at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
at com.arasthel.swissknife.SwissKnife.inject(SwissKnife.groovy:40)
at java.lang.reflect.Method.invoke(Native Method)
at java.lang.reflect.Method.invoke(Method.java:372)
at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
at org.codehaus.groovy.runtime.callsite.StaticMetaMethodSite$StaticMetaMethodSiteNoUnwrapNoCoerce.invoke(StaticMetaMethodSite.java:151)
at org.codehaus.groovy.runtime.callsite.StaticMetaMethodSite.call(StaticMetaMethodSite.java:91)
at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
at com.tutosandroidfrance.wearsample.MainActivity.onCreate(MainActivity.groovy:45)

If I take out the inject method call it works fine (but my injected views are still null).

Here is the project with full source:
https://github.com/rvanderwerf/WearDavinciEmmetGroovy

**Comment by Arasthel** [comment]
For some reason `@InjectView` transformation isn't working at all, which doesn't really make sense. I'm testing it a bit, but right now I don't really know what causes this.

**Comment by rvanderwerf** [comment]
thanks for taking a look!

**Comment by Arasthel** [comment]
@eugene-kamenev and I are still trying to figure out what's wrong, but we've got nothing so far. 

I just copied SK and its wear modules to your project and copied all your module dependencies to the sample wear, and it works but yours don't, it doesn't matter what we do...

Right now, their gradle files are almost identical, I tried disabling multidex, gradle optimizations and tried a few more ideas, but nothing really changed. For some reason your module refuses to call the AST transform methods.

**Comment by rvanderwerf** [comment]
oh course it would be mine! what about the compiletarget of 22 version maybe?

**Comment by rvanderwerf** [comment]
well I have some news on the bug. I took out the GridViewPager and DotsPageIndicator stuff, and SwissKnife works now. I have a branch I made to remote control a Lego EV3 robot and the injection and button even stuff works now. So I guess it seems to be a compatability thing with GridViewPager as far as I can tell.
this is the branch I have where SK is now working https://github.com/rvanderwerf/WearDavinciEmmetGroovy/tree/EV3Wear

**Comment by Arasthel** [comment]
Well, I do know what the bug was...

You had this:

```groovy
private GridViewPager pager
private DotsPageIndicator dotsPageIndicator
...
@InjectView(value=R.id.pager)
GridViewPager pager

@InjectView(value=R.id.page_indicator)
DotsPageIndicator dotsPageIndicator
```
The same field was added twice. I suppose groovy detected this somehow and instead of crashing just ignored the 2nd declaration, so the injection annotations were never processed and nothing really happened, I just tried this and it works as soon as you delete the private fields.

I'm closing this as it wasn't really a bug.
